com.unity.multiplayer.mlapi/Runtime/Core/NetworkBehaviour.cs

```diff
@@ -46,34 +59,36 @@ internal NetworkSerializer __beginSendServerRpc(uint rpcMethodId, ServerRpcParam
         {
             PooledNetworkWriter writer;
 
-            var rpcQueueContainer = NetworkManager.RpcQueueContainer;
-            var isUsingBatching = rpcQueueContainer.IsUsingBatching();
+            SetUpdateStage(ref serverRpcParams.Send);
+
+            if (serverRpcParams.Send.UpdateStage == NetworkUpdateStage.Initialization)
+            {
+                throw new NotSupportedException(
+                    $"{nameof(NetworkUpdateStage.Initialization)} cannot be used as a target for processing RPCs.");
+            }
+
+            var messageQueueContainer = NetworkManager.MessageQueueContainer;
+            var isUsingBatching = messageQueueContainer.IsUsingBatching();
             var transportChannel = rpcDelivery == RpcDelivery.Reliable ? NetworkChannel.ReliableRpc : NetworkChannel.UnreliableRpc;
 
             if (IsHost)
             {
-                writer = rpcQueueContainer.BeginAddQueueItemToFrame(RpcQueueContainer.QueueItemType.ServerRpc, Time.realtimeSinceStartup, transportChannel,
```

Comment:
Probably too late for this PR, but I would say for the next one, when we have a rename like this that touches so many things, let's do that rename in one PR and then rebase this one off of that